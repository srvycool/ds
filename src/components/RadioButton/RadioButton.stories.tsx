import React, { useState } from 'react';
import { RadioButton } from './RadioButton';

export default {
  title: 'RadioButton',
  component: RadioButton,
};

export const Default: React.FC = () => {
  const [value, setValue] = useState('MULTIPLE_CHOICE');

  return (
    <form>
      <RadioButton
        headline="Multiple Choice"
        description="Elit in fugiat minim do"
        onChange={(event) => setValue(event.target.value)}
        checked={value === 'MULTIPLE_CHOICE'}
        value="MULTIPLE_CHOICE"
        marginBottom="m"
        name="answer"
      />
      <RadioButton
        headline="Text Field"
        description="Elit in fugiat minim do"
        checked={value === 'TEXTFIELD'}
        value="TEXTFIELD"
        name="answer"
        onChange={(event) => setValue(event.target.value)}
      />
    </form>
  );
};
