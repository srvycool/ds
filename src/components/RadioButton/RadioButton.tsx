import React from 'react';
import styled, { StyledComponentProps } from 'styled-components';
import { Box } from '../Box/Box';
import { Theme } from '../../theme';
import { MarginBottomProps } from 'styled-system';
import { Typography } from '../Typography/Typography';

export type RadioButtonProps = StyledComponentProps<'input', Theme, {}, never> &
  MarginBottomProps<Theme> & {
    headline: string;
    description: string;
  };

const StyledRadioButton = styled(Box)`
  cursor: pointer;
  transition: 100ms border-color ease-in-out;

  &:focus-within {
    border-color: ${(props) => props.theme.colors.primary500};
  }
`;

const StyledRadio = styled.div<{ checked: boolean }>`
  width: 24px;
  height: 24px;
  border-radius: 100%;
  border: 4px solid ${(props) => props.theme.colors.grey900};
  background-color: ${(props) => props.theme.colors.grey000};
  display: flex;
  justify-content: center;
  align-items: center;
  flex-grow: 0;
  flex-shrink: 0;

  &:after {
    content: '';
    display: ${(props) => (props.checked ? 'block' : 'none')};
    width: 12px;
    height: 12px;
    background-color: ${(props) => props.theme.colors.grey900};
    border-radius: 100%;
  }
`;

const StyledInput = styled.input`
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  z-index: -1;
`;

export const RadioButton: React.FC<RadioButtonProps> = ({
  headline,
  description,
  marginBottom,
  ...props
}) => {
  return (
    <StyledRadioButton
      as="label"
      display="flex"
      flexDirection="row"
      alignItems="center"
      position="relative"
      paddingTop="m"
      paddingBottom="m"
      paddingRight="l"
      paddingLeft="l"
      boxShadow="primary"
      border="primary"
      borderRadius="primary"
      width="100%"
      marginBottom={marginBottom}
    >
      <StyledInput type="radio" {...props} />
      <StyledRadio checked={props.checked || false} />
      <Box as="span" display="block" paddingLeft="l">
        <Typography variant="body1" color="grey900" display="block">
          {headline}
        </Typography>
        <Typography variant="body2" color="grey800">
          {description}
        </Typography>
      </Box>
    </StyledRadioButton>
  );
};
