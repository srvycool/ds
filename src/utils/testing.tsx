import React from 'react';
import {
  render as tlrender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { ThemeProvider } from '../components/ThemeProvider/ThemeProvider';

export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult {
  return tlrender(<ThemeProvider>{ui}</ThemeProvider>, options);
}
