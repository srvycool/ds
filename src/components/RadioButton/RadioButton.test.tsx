import React from 'react';
import { render } from '../../utils/testing';
import { RadioButton } from './RadioButton';

describe('RadioButton', () => {
  describe('checked', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <RadioButton
          headline="Et occaecat aliquip"
          description="Elit in fugiat minim do"
          checked={true}
          value="test"
          name="test"
          onChange={() => null}
        />
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('not checked', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <RadioButton
          headline="Et occaecat aliquip"
          description="Elit in fugiat minim do"
          checked={false}
          value="test"
          name="test"
          onChange={() => null}
        />
      );

      expect(container).toMatchSnapshot();
    });
  });
});
