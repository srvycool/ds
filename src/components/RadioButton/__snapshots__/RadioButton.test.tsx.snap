// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton checked should match snapshot 1`] = `
.c3 {
  display: block;
  padding-left: 20px;
  box-sizing: border-box;
}

.c4 {
  margin: 0;
  color: #493C3F;
  display: block;
  font-size: 16px;
  line-height: 1.2;
  font-family: Montserrat;
}

.c5 {
  margin: 0;
  color: #7D666C;
  font-size: 12px;
  line-height: 1.2;
  font-family: Montserrat;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-right: 20px;
  padding-left: 20px;
  box-shadow: 0px 1px 2px rgba(0,0,0,0.06),0px 1px 3px rgba(0,0,0,0.1);
  border: 2px solid #E6DFE1;
  border-radius: 8px;
  width: 100%;
  box-sizing: border-box;
  cursor: pointer;
  -webkit-transition: 100ms border-color ease-in-out;
  transition: 100ms border-color ease-in-out;
}

.c0:focus-within {
  border-color: #DA4167;
}

.c2 {
  width: 24px;
  height: 24px;
  border-radius: 100%;
  border: 4px solid #493C3F;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:after {
  content: '';
  display: block;
  width: 12px;
  height: 12px;
  background-color: #493C3F;
  border-radius: 100%;
}

.c1 {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  z-index: -1;
}

<div>
  <label
    class="c0"
    display="flex"
    width="100%"
  >
    <input
      checked=""
      class="c1"
      name="test"
      type="radio"
      value="test"
    />
    <div
      class="c2"
    />
    <span
      class="c3"
      display="block"
    >
      <span
        class="c4"
        color="grey900"
        display="block"
      >
        Et occaecat aliquip
      </span>
      <span
        class="c5"
        color="grey800"
      >
        Elit in fugiat minim do
      </span>
    </span>
  </label>
</div>
`;

exports[`RadioButton not checked should match snapshot 1`] = `
.c3 {
  display: block;
  padding-left: 20px;
  box-sizing: border-box;
}

.c4 {
  margin: 0;
  color: #493C3F;
  display: block;
  font-size: 16px;
  line-height: 1.2;
  font-family: Montserrat;
}

.c5 {
  margin: 0;
  color: #7D666C;
  font-size: 12px;
  line-height: 1.2;
  font-family: Montserrat;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-right: 20px;
  padding-left: 20px;
  box-shadow: 0px 1px 2px rgba(0,0,0,0.06),0px 1px 3px rgba(0,0,0,0.1);
  border: 2px solid #E6DFE1;
  border-radius: 8px;
  width: 100%;
  box-sizing: border-box;
  cursor: pointer;
  -webkit-transition: 100ms border-color ease-in-out;
  transition: 100ms border-color ease-in-out;
}

.c0:focus-within {
  border-color: #DA4167;
}

.c2 {
  width: 24px;
  height: 24px;
  border-radius: 100%;
  border: 4px solid #493C3F;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:after {
  content: '';
  display: none;
  width: 12px;
  height: 12px;
  background-color: #493C3F;
  border-radius: 100%;
}

.c1 {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  z-index: -1;
}

<div>
  <label
    class="c0"
    display="flex"
    width="100%"
  >
    <input
      class="c1"
      name="test"
      type="radio"
      value="test"
    />
    <div
      class="c2"
    />
    <span
      class="c3"
      display="block"
    >
      <span
        class="c4"
        color="grey900"
        display="block"
      >
        Et occaecat aliquip
      </span>
      <span
        class="c5"
        color="grey800"
      >
        Elit in fugiat minim do
      </span>
    </span>
  </label>
</div>
`;
